File: Backend/Financeiro.ConciliacaoWebAPI/JsonWebToken/Code/Intervalo/IntervaloBLL.cs
using JsonWebToken.Code.Intervalo;
using JsonWebToken.Model;
using System;
using System.Collections.Generic;

namespace JsonWebToken.Code
{
  public class IntervaloBLL
  {

    public IntervaloModel selectIntervaloPorId(int id)
    {
      try
      {
        IntervaloDAL DAL = new IntervaloDAL();
        return DAL.selectIntervaloPorId(id);
      }
      catch (Exception e)
      {
        throw e;
      }
    }
    public List<IntervaloModel> selectIntervalo(int periodo)
    {
      try
      {
        IntervaloDAL DAL = new IntervaloDAL();
        List<IntervaloModel> list = DAL.selectIntervalo(periodo);
        return list;
      }
      catch (Exception e)
      {
        throw e;
      }
    }


    public void deleteIntervalo(int id)
    {
      try
      {
        IntervaloDAL DAL = new IntervaloDAL();
        DAL.deleteIntervalo(id);

      }
      catch (Exception e)
      {
        throw e;
      }
    }

    public void updateIntervalo(IntervaloModel intervalo)
    {
      try
      {
        IntervaloDAL DAL = new IntervaloDAL();
        DAL.updateIntervalo(intervalo);

      }
      catch (Exception e)
      {
        throw e;
      }
    }

    public void insertIntervalo(IntervaloModel intervalo)
    {
      try
      {
        IntervaloDAL DAL = new IntervaloDAL();
        DAL.inserirIntervalo(intervalo);

      }
      catch (Exception e)
      {
        throw e;
      }

    }
    public void ativarDesativarIntervalo(IntervaloModel Intervalo)
    {
      try
      {
        IntervaloDAL DAL = new IntervaloDAL();
        if (Intervalo.situacao.Equals("0"))
        {
          DAL.ativarIntervalo(Intervalo.codIntervalo);
        }
        else
        {
          DAL.desativarIntervalo(Intervalo.codIntervalo);
        }
      }
      catch (Exception e)
      {
        throw e;
      }

    }
  }
}

File: Backend/Administracao.ConciliacaoAPI/Code/Intervalo/IntervaloBLL.vb
Imports Administracao.ConciliacaoAPI.Model

Namespace Code.Intervalo
    Public Class IntervaloBLL

        Public Function selectIntervaloPorId(id As Integer) As IntervaloModel
            Try
                Dim DAL As New IntervaloDAL()
                Return DAL.selectIntervaloPorId(id)
            Catch e As Exception
                Throw e
            End Try
        End Function
    Public Function selectIntervalo(periodo As Integer, situacao As String) As List(Of IntervaloModel)
      Try
        Dim DAL As New IntervaloDAL()
        Dim list As List(Of IntervaloModel) = DAL.selectIntervalo(periodo, situacao)
        Return list
      Catch e As Exception
        Throw e
      End Try
    End Function


    Public Sub deleteIntervalo(id As Integer)
            Try
                Dim DAL As New IntervaloDAL()

                DAL.deleteIntervalo(id)
            Catch e As Exception
                Throw e
            End Try
        End Sub

        Public Sub updateIntervalo(intervalo As IntervaloModel)
            Try
                Dim DAL As New IntervaloDAL()

                DAL.updateIntervalo(intervalo)
            Catch e As Exception
                Throw e
            End Try
        End Sub

        Public Sub insertIntervalo(intervalo As IntervaloModel)
            Try
                Dim DAL As New IntervaloDAL()

                DAL.inserirIntervalo(intervalo)
            Catch e As Exception
                Throw e
            End Try

        End Sub

        Public Function selectIntervaloDeAte(inicio As Integer, fim As Integer, periodo As Integer) As Integer
            Try
                Dim DAL As New IntervaloDAL()

                Return DAL.selectIntervaloDeAte(inicio, fim, periodo)
            Catch e As Exception
                Throw e
            End Try

        End Function

        Public Sub ativarDesativarIntervalo(Intervalo As IntervaloModel)
            Try
                Dim DAL As New IntervaloDAL()
                If Intervalo.situacao.Equals(0) Then
                    DAL.ativarIntervalo(Intervalo.codIntervalo)
                Else
                    DAL.desativarIntervalo(Intervalo.codIntervalo)
                End If
            Catch e As Exception
                Throw e
            End Try

        End Sub
    End Class
End Namespace
